docs/general/options.md

```diff
@@ -0,0 +1,48 @@
+# Options
+
+## Scriptable Options
+
+Scriptable options also accept a function which is called for each data and that takes the unique argument `context` representing contextual information (see [option context](options.md#option-context)).
+
+Example:
+
+```javascript
+color: function(context) {
```

Comment:
I initially implemented scriptable options as a function taking 2 arguments: an `index` and the `context` (see the [datalabels plugin](https://github.com/chartjs/chartjs-plugin-datalabels/blob/master/docs/options.md#scriptable-options)) - `index` being the dataset index if the option is resolved for a dataset element (line of a line chart) or the data index if the option is for a data element (point of a line chart).

Having the index as first argument would allow to integrate with external functions that (e.g. returning a color for a given index (`return index % size`)). After discussing with @etimberg, we agreed to remove that first argument and keep only the `context`.

Thoughts?